src/rf/js/src/home/components/modals.js

```diff
@@ -107,9 +188,224 @@ var UploadModal = React.createBackboneClass({
+
+        if (responseJSON.errors.duplicate_name) {
+            errors.duplicate_name = 'Please choose a name not already in use.';
+        }
+
+        return errors;
+    },
+
+    getUpdatedServerErrors: function(oldState) {
+        // serverErrors should be removed if the input
+        // value that led to the error has been changed.
+        // Of course, the new value could still lead to a server error later.
+        var errors = this.state.serverErrors;
+        if (oldState.name != this.state.name) {
+            errors.duplicate_name = '';
+        }
+        return errors;
+    },
+
+    goToPane: function(paneInd) {
+        this.setState({activePane: paneInd});
+    },
+
+    backPane2: function(e) {
+        e.preventDefault();
+        this.goToPane(1);
+    },
+
+    backPane3: function(e) {
+        e.preventDefault();
+        this.goToPane(2);
+    },
+
+    updateState: function(callback) {
+        // Update the state using values in input elements,
+        // update client and server errors, and then call
+        // optional callback.
+        var self = this,
+            oldState = this.state;
+
+        self.setState(self.getLayerFields(), function() {
+            self.setState({clientErrors: self.getClientErrors()}, function() {
+                self.setState({serverErrors: self.getUpdatedServerErrors(oldState)},
+                              callback || _.noop);
+            });
+        });
+    },
+
+    attemptContinuePane1: function(e) {
+        e.preventDefault();
+        var self = this;
+        this.updateState(function() {
+            if (self.pane1Valid()) {
+                self.goToPane(2);
+                self.setState({showPane1ClientErrors: false});
+            } else {
+                self.setState({showPane1ClientErrors: true});
+            }
+        });
+    },
+
+    attemptContinuePane2: function(e) {
+        e.preventDefault();
+        var self = this;
+        this.updateState(function() {
+            if (self.pane2Valid()) {
+                self.goToPane(3);
+                self.setState({showPane2ClientErrors: false});
+            } else {
+                self.setState({showPane2ClientErrors: true});
+            }
+        });
+    },
+
+    attemptSubmit: function(e) {
+        e.preventDefault();
+        var self = this;
+        this.updateState(function() {
+            self.postLayer()
+                .done(function() {
+                    self.uploadFiles();
+                    self.clear();
+                    $('.import-modal').modal('hide');
+                })
+                .fail(function(e) {
+                    self.goToPane(2);
+                    self.setState({serverErrors: self.getServerErrors(e.responseJSON)});
+                });
+        });
+    },
+
+    postLayer: function() {
+        var username = settings.getUser().get('username'),
+            url = '/user/' + username + '/layer/create',
```

Comment:
We might consider adding this URL to the user details model so we don't have to reverse the URL here. See this for an example https://github.com/azavea/raster-foundry/pull/97/files#diff-86968bd9012a2dccfcbe09fdc6ccb342R45.